ports/servoshell/desktop/dialog.rs

```diff
@@ -45,55 +47,74 @@ impl Dialog {
                 .default_file_filter("All Supported Types");
         }
 
-        let dialog = FileDialog {
+        Dialog::File {
             dialog,
             multiple,
             response_sender,
-        };
+        }
+    }
 
-        Dialog::File(dialog)
+    pub fn new_alert_dialog(message: String, sender: IpcSender<()>) -> Self {
+        Dialog::Alert { message, sender }
     }
 
     pub fn update(&mut self, ctx: &egui::Context) -> bool {
```

Comment:
Maybe we can add documentation here about return value i.e `true` means the dialog is still open shouldn't be dropped and `false` implies the dialog has been closed or cancelled.